api/src/main/resources/workbench.yaml

```diff
@@ -2964,19 +2964,45 @@ definitions:
         type: object
         description: Can be any value
 
-  DataSetPreviewResponse:
+  DataSetPreviewRequest:
     type: object
+    required:
+      - domain
     properties:
-      domainValue:
-        type: "array"
+      domain:
+        $ref: "#/definitions/Domain"
+      includesAllParticipants:
+        type: boolean
+        default: false
+        description: >
+          Whether to include all participants or filter by cohorts
+      prePackagedConceptSet:
+        $ref: "#/definitions/PrePackagedConceptSetEnum"
+      conceptSetIds:
+        type: array
+        description: >
+          The ids of all concept sets in the data set
         items:
-          $ref: "#/definitions/DataSetPreviewList"
+          type: integer
+          format: int64
+      cohortIds:
+        type: array
+        description: >
+          The ids of all cohorts in the data set
+        items:
+          type: integer
+          format: int64
+      values:
+        type: array
+        description: >
+          All the selected values in the data set
+        items:
+          type: string
 
-  DataSetPreviewList:
+
+  DataSetPreviewResponse:
     type: object
     properties:
-      domain:
```

Comment:
Semantically, I'm not sure if this should return a domain. It should be deterministic from the request sent up, so should be redundant information.

Reply:
The domain is not in `DataSetPreviewValueList`, so I will add domain back in.